common/browser_context.go

```diff
@@ -73,38 +74,74 @@ func (b *BrowserContext) AddCookies(cookies goja.Value) *goja.Promise {
 func (b *BrowserContext) AddInitScript(script goja.Value, arg goja.Value) {
 	b.logger.Debugf("BrowserContext:AddInitScript", "bctxid:%v", b.id)
 
-	rt := b.vu.Runtime()
+	if script == nil || goja.IsUndefined(script) || goja.IsNull(script) {
+		k6ext.Panic(b.ctx, "parsing init script: invalid script")
+	}
```

Comment:
Here, you should return an error and use the helper `gojaValueExists`. Please see the PRs of the sub-issues of #683 for more details (for returning errors and mapping them in the mapping layer).

Reply:
So if I understand correctly, that implies making the `addInitScript` method return a promise in the JS API right?

Reply:
Implemented in [a3c7e31](https://github.com/grafana/xk6-browser/pull/719/commits/a3c7e31431db1363bb590441170488e1ddb2b4ee).